assets/js/language.js

```diff
@@ -0,0 +1,67 @@
+// Modified from https://booky.dev/assets/js/language.js
+var langCache = {}
+
+const getLanguage = () => {
+	if (getCookie("lang")) return getCookie("lang");
+
+	const userLang = navigator.language || navigator.userLanguage;
+	return userLang ? (userLang.split("-")[0] == "de" ? "de" : "en") : "en";
+};
+
+var reloadText = async language => {
+	setCookie("lang", language, 60, true);
+	document.documentElement.lang = language;
+
+	const i18n = document.querySelectorAll("[translation]");
+	if (i18n.length <= 0) return;
+
+	const scrollTop = document.documentElement.scrollTop || document.body.scrollTop;
+
+	const json = await loadLangFile(language)
+	for (let i = 0; i < i18n.length; i++) {
+		const element = i18n.item(i);
+		const key = element.getAttribute("translation");
+		const splitted = key.split(".");
+
+		var text = "";
+		if (json[splitted[0]][splitted[1]]) {
+		    if (json[splitted[0]][splitted[1]][splitted[2]]) {
+		    	if (json[splitted[0]][splitted[1]][splitted[2]][splitted[3]]) text = json[splitted[0]][splitted[1]][splitted[2]][splitted[3]];
+		    	else text = json[splitted[0]][splitted[1]][splitted[2]];
+		    } else text = json[splitted[0]][splitted[1]];
+		} else text = json[splitted[0]];
```

Comment:
This looks like bullshit (because it is). Use something like this:
```js
const resolveValue = (obj, key) => {
  return resolveValue0(obj, key.split("."));
};

const resolveValue0 = (obj, keySplit) => {
  if (keySplit.length == 1) return obj[keySplit[0]];
  return resolveValue0(obj[keySplit[0]], keySplit.slice(1));
};
```
(two functions, call "`resolveValue(json, key)`")